### PR TITLE
feature: endpoint search

### PR DESCRIPTION
**Pull request by Abrcobgal**

### CODE REVIEW

**Review by andrsdt — CHANGES_REQUESTED** [review]
This is unusable in the front and does not comply with any REST standards, please make the requested changes

**Review by andrsdt — CHANGES_REQUESTED** [review]
* min_stars doesn't work, if I set min_stars=9999 I get results when there is no user with a score of 9999 (maximum is 5.0)

* date_from and date_to don't work, if I filter by dates between 2030-03-07 and 2030-03-08 I get back trips when in reality there are none

* hour_from and hour_to don't work, if I set the time between 00:00 and 00:01 I get back trips when in fact there are none

* min_price and max_price should be inclusive (max 4€ should include the 4€ ones). I have already fixed this.

File: bugalink-backend/trips/views.py [hunk 1]
```diff
@@ -81,3 +100,118 @@ def reject(self, request, *args, **kwargs):
...
+        prefers_music,
+        prefers_talk,
+        allows_pets,
+        allows_smoking,
+        *args,
+        **kwargs
+    ):
+        try:
+            # Se busca entre los viajes pendientes
+            trips = Trip.objects.filter(status="PENDING")
```
[the remark this answers is not in the document]
Comment (Abrcobgal) [reply bugalink-backend/trips/views.py:118]: Enrique will take this as his own issue, it will be like that for now as far as it works

**Review by andrsdt — APPROVED** [review]
I have made some big changes in the method that decreased the complexity significantly. Take note for the next time when you see a structure repeated many times (there were +10 if statements, all of them with the same structure). The code is much cleaner now.

image

I also fixed a bug that happened when there was only one trip left in the filter (it would never get removed) due to a misconception in python's mutable list capabilities. An example with many filters:

image

**Pull request merged**